xarray/core/common.py

```diff
@@ -576,7 +576,13 @@ def where(self, cond, other=None, drop=False):
             outobj = self
             outcond = cond
 
-        return outobj._where(outcond)
+        # preserve attributes
+        out = outobj._where(outcond)
+        out.attrs = self.attrs
+        if isinstance(out, Dataset):
```

Comment:
Can you implement a private method defined on both `Dataset`/`DataArray`, something like `_copy_attrs_from(self, other)`? Then we can call `out._copy_attrs_from(self)` for a generic solution.